botocore/client.py

```diff
@@ -832,27 +857,29 @@ def _resolve_signature_version(self, service_name, resolved):
         if configured_version is not None:
             return configured_version
 
-        potential_versions = resolved.get('signatureVersions', [])
-        if (
-            self.service_signature_version is not None
-            and self.service_signature_version
-            not in _LEGACY_SIGNATURE_VERSIONS
-        ):
-            # Prefer the service model as most specific
-            # source of truth for new signature versions.
-            potential_versions = [self.service_signature_version]
+        # These have since added the "auth" key to the service model
+        # with "aws.auth#sigv4", but preserve existing behavior from
+        # when we preferred endpoints.json over the service models
+        if service_name in ['s3', 's3-control']:
```

Comment:
nit; Static lists we should do as tuples instead of lists ideally so we're not allocating unnecessary memory.
```suggestion
        if service_name in ('s3', 's3-control'):
```